### PR TITLE
Add functionality LED Sign page saves previous input

### LINKED ISSUE #380: Show Existing Messages on LED Sign
### setup to develop
- `python setup.py`
- `npm install && npm run server-install`

### development
1. run frontend with `npm run frontend`
1. run mongod (either in terminal w `mongod` or application)
1. run server with `npm run server`
1. run `led_sign_server.py` in printingRPC
1. in `api/config/config` change the IP adress to `localhost`

### Steps
- update the sign text using the webpage
- start by doing a `console.log` of the  status returned from the sign [here](https://github.com/SCE-Development/Core-v4/blob/fe1b1d56d8bffd86e5a41425bdb11fa59bf0bded/src/Pages/LedSign/LedSign.js#L94)
- take a look at the values in the object that was printed and use it to udpate the UI

### PR DESCRIPTION
**Pull request by hsorathia**
Resolves #380

### CODE REVIEW

File: src/Pages/LedSign/LedSign.js [hunk 1]
```diff
@@ -87,13 +89,22 @@ function LedSign(props) {
...
         setSignHealthy(true);
+        if(status.responseData && status.responseData.text) {
+          setText(status.responseData.text);
+          setBrightness(status.responseData.brightness);
+          setScrollSpeed(status.responseData.scrollSpeed);
+          setBackgroundColor(status.responseData.backgroundColor);
+          setTextColor(status.responseData.textColor);
+          setBorderColor(status.responseData.borderColor);
+        }
+        // console.log('we done waiting');
```
Comment (jerrylee17) [inline src/Pages/LedSign/LedSign.js:107]: delete comments

File: src/Pages/LedSign/LedSign.js [hunk 2]
```diff
@@ -87,13 +89,22 @@ function LedSign(props) {
...
   }
-
+  
   useEffect(() => {
     async function checkSignHealth() {
       setLoading(true);
       const status = await healthCheck(props.user.firstName);
       if (status && !status.error) {
         setSignHealthy(true);
+        if(status.responseData && status.responseData.text) {
```
Comment (jerrylee17) [inline src/Pages/LedSign/LedSign.js:99]: Is this statement necessary? Because isn't there always going to be responseData?

Comment (hsorathia) [reply src/Pages/LedSign/LedSign.js:99]: Yeah, it's to deal with the first time the sign gets response data from the led server. If the server gives blank information this if statement is there to ignore it and use the initial states of the hooks.

File: src/Pages/LedSign/LedSign.js [hunk 3]
```diff
@@ -87,13 +88,20 @@ function LedSign(props) {
       );
     }
   }
-
   useEffect(() => {
     async function checkSignHealth() {
       setLoading(true);
       const status = await healthCheck(props.user.firstName);
       if (status && !status.error) {
         setSignHealthy(true);
```
Comment (evanugarte) [inline src/Pages/LedSign/LedSign.js:96]: i see a lot of `status.responseData` in the lines below. consider doing this:

```js
const { responseData } = status;
if (responseData && responseData.text) {
  const { text, brightness, scrollSpeed, backgroundColor, /* etc */ } = responseData;
  setText(text)
  setBrightness(brightness)
  /* etc */
```

File: test/LedSign.js [hunk 4]
```diff
@@ -39,6 +37,17 @@ const VALID_SIGN_REQUEST = {
   firstName: 'John Doe',
   email: 'bigoof@gmail.com'
 };
+const SUCCESS_MESSAGE = {
+  message: {
+    getText:  () => {return VALID_SIGN_REQUEST.text;},
+    getBrightness: () => {return VALID_SIGN_REQUEST.brightness;},
```
Comment (evanugarte) [inline test/LedSign.js:43]: this is fire dude, great job!!!!

File: test/LedSign.js [hunk 5]
```diff
@@ -62,19 +71,26 @@ describe('LedSign', () => {
...
           expect(res).to.have.status(OK);
           done();
         })
         .catch(err => {
           throw err;
         });
     });
+    it('should return the correct values when modified', done => {
+      healthCheckMock.resolves(SUCCESS_MESSAGE);
+      expect(signResponse.text).to.equal(VALID_SIGN_REQUEST.text);
```
Comment (evanugarte) [inline test/LedSign.js:91]: i dont think you finished here, add
```js
expect(signResponse.brightness).to.equal(VALID_SIGN_REQUEST.brightness);
expect(signResponse.scrollSpeed).to.equal(VALID_SIGN_REQUEST.scrollSpeed);
/* do the rest below */
```

File: test/LedSign.js (hunk 4 above)
Comment (hsorathia) [reply test/LedSign.js:43]: 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯 💯

**Pull request merged**